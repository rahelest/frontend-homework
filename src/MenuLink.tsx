type Props = {
  text: string,
  icon: string,
}

const MenuLink = ({ icon, text }: Props) => (
  <div>
    <i className="material-icons-outlined">
      {icon}
    </i>

    {text}
  </div>
)

export default MenuLink
