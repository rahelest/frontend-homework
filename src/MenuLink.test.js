import { shallow } from 'enzyme'
import MenuLink from './MenuLink'

describe('<MenuLink />', () => {
  const render = (props = {}) =>
    shallow(<MenuLink icon="phone" text="Call me" href="/abc" {...props} />)

  it('renders link', () => {
    expect(render()).toMatchSnapshot()
  })

  it('renders link with a provided target', () => {
    expect(render({ target: '_blank' })).toMatchSnapshot()
  })
})
