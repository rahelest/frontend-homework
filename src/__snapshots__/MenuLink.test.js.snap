// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> renders link 1`] = `
<div>
  <i
    className="material-icons-outlined"
  >
    phone
  </i>
  Call me
</div>
`;
